### PR TITLE
Reformat for clarity, add tests and prevent trigger loops

### PR DESCRIPTION
**Pull request by jjcastro**
Reformat main code for clarity and to enable it to be used in tests.

Also prevent main function body from running if the cloud function is triggered via a document update to outputs (only new documents or changes to inputs)

Add initial tests for this too. More tests incoming, created json_test_data for this purpose.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -18,42 +20,150 @@ def hello_firestore(cloud_event: CloudEvent) -> None:
...
+    water_pct, land_use_anthropic_pct, land_use_primary_vegetation_pct, land_use_secondary_vegetation_or_regrowth_pct = origin_enhancement(float(value.get('lat')), float(value.get('lon')))
+
+    value['water_pct'] = water_pct
+    value['land_use_anthropic_pct'] = land_use_anthropic_pct
+    value['land_use_primary_vegetation_pct'] = land_use_primary_vegetation_pct
+    value['land_use_secondary_vegetation_or_regrowth_pct'] = land_use_secondary_vegetation_or_regrowth_pct
+    
+    return value
+
+def fraud_detection_process_sample(value: dict):
```
[the remark this answers is not in the document]
Comment (benwulfe) [reply main.py:136]: The owners of each of these decorators can then add their own unit tests

File: tests/test_main.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            "test": "test"
+        }
+        new_value = {
+            "test": "test",
+            "validity_details": "test"
+        }
+        result = is_document_creation_or_update_with_new_inputs(old_value, new_value)
+        self.assertFalse(result)
+
+    def test_is_document_creation_or_update_with_new_inputs___document_update_with_new_inputs(self):
```
Comment (benwulfe) [inline tests/test_main.py:31]: split into a test per decorator.
Also, then we can have each decorator add its own html/pdf snippet.

So a report decorator will have:
* a value decorator
* unit tests
* pdf/html snippet generator

**Review by benwulfe — CHANGES_REQUESTED** [review]
Please split out the calculations and then it looks good to me.

File: main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jjcastro) [reply main.py]: Moved into their own files.

File: tests/test_main.py (hunk 2 above)
Comment (jjcastro) [reply tests/test_main.py:31]: These are just filters to run the validation though, so i think it makes sense to keep them in main and therefore in the main test. WDYT?

**Review by benwulfe — APPROVED** [review]
looks really good.  please run by kader who is changing stuff on the cloud function side and may have an inflight change that needs to be merged with this.

**Pull request merged**